src/base/http/requestparser.cpp

```diff
@@ -59,21 +59,19 @@ namespace
         return in;
     }
 
-    bool parseHeaderLine(const QStringView line, HeaderMap &out)
+    std::tuple<QString, QString> parseHeaderLine(const QByteArrayView line)
     {
         // [rfc7230] 3.2. Header Fields
         const int i = line.indexOf(u':');
         if (i <= 0)
         {
             qWarning() << Q_FUNC_INFO << "invalid http header:" << line;
-            return false;
+            return {};
         }
 
-        const QString name = line.first(i).trimmed().toString().toLower();
-        const QString value = line.sliced(i + 1).trimmed().toString();
-        out[name] = value;
-
-        return true;
+        const QString name = QString::fromLatin1(line.first(i).trimmed()).toLower();
```

Comment:
Btw, I really hate that `.toLower()` creates a copy.

What if we introduce a new utility function to lower strings in place? It's as simple as

```cpp
void Utils::String::toLower(QString &string) {
    for(QChar &c : string)
        c = c.toLower();
}
```

Reply:
If converting to lowercase is required only for comparison purposes, I'd rather leave the name as it is and use a case-insensitive comparison instead.

Reply:
Header names used as map keys.

https://github.com/qbittorrent/qBittorrent/blob/7a1a214f73f99fdf81aac8078429525c120df4f1/src/base/http/types.h#L106

So no, case conversion is crucial to extract values from the map.

Reply:
> obviously `HEADER_X_FORWARDED_PROTO` here is supposed to be all lowercase as everything else.

Good catch 👍

Reply:
> Although, I'm not proficient in C++ to claim that compiler could not optimize it out.

This is not about the compiler, but about an explicitly provided optimized overload for a temporary object.